File: app/src/main/java/com/originb/inkwisenote2/config/AppConfig.java
package com.originb.inkwisenote2.config;

import com.originb.inkwisenote2.modules.handwrittennotes.data.PageTemplate;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Getter
@Setter
public class AppConfig {
    private List<Feature> enabledFeatures;
    private Map<String, PageTemplate> pageTemplates;
    private List<CanvasSize> canvasSizes;
    private AppSecrets appSecrets;

    private Map<ConfigKeys, String> runtimeSettings;

    public AppConfig() {
        enabledFeatures = new ArrayList<>();
        pageTemplates = new HashMap<>();
        appSecrets = new AppSecrets();
        runtimeSettings = new HashMap<>();
        canvasSizes = new ArrayList<>();
    }

    public static AppConfig createDefault() {
        return new AppConfig();
    }

}

File: app/src/main/java/com/originb/inkwisenote2/config/AppConfig.kt
package com.originb.inkwisenote2.config

import com.originb.inkwisenote2.modules.handwrittennotes.data.PageTemplate

class AppConfig {
    var enabledFeatures: MutableList<Feature?>? = ArrayList()
    private val pageTemplates: MutableMap<String?, PageTemplate?>? = HashMap()
    private val canvasSizes: MutableList<CanvasSize?>? = ArrayList()
    var appSecrets: AppSecrets? = AppSecrets()
    private val runtimeSettings: MutableMap<ConfigKeys?, String?>? = HashMap()

    fun getRuntimeSettings(): MutableMap<ConfigKeys?, String?>? = runtimeSettings
    fun getPageTemplates(): MutableMap<String?, PageTemplate?>? = pageTemplates
    fun getCanvasSizes(): MutableList<CanvasSize?>? = canvasSizes

    companion object {
        fun createDefault(): AppConfig {
            return AppConfig()
        }
    }
}

